server/src/main/java/org/opensearch/transport/client/support/AbstractClient.java

```diff
@@ -2145,6 +2146,11 @@ public void updateView(CreateViewAction.Request request, ActionListener<GetViewA
         public ActionFuture<GetViewAction.Response> updateView(CreateViewAction.Request request) {
             return execute(UpdateViewAction.INSTANCE, request);
         }
+
+        public ScaleIndexRequestBuilder prepareScaleSearchOnly(String index) {
```

Comment:
Related to my comment above about the `ScaleIndexRequestBuilder` and whether there should be a default, I'm more convinced by this method that we should force clients to select the direction of scaling. 

Right now, someone would call `prepareScaleSearchOnly` and get a `ScaleIndexRequestBuilder` that **scales up  from search only**, by default.

Reply:
Sure Froh, let me update as `prepareScaleSearchOnly(String index, boolean searchOnly)` and so the `ScaleIndexRequestBuilder(OpenSearchClient client, boolean searchOnly, String index), this was we can have three-arg constructor and remove the defaults.